runner.py

```diff
@@ -0,0 +1,51 @@
+import runner_run_funcs
+import json
+
+
+class Runner:
+
+    run_func_ltable, data_gen_func_ltable = runner_run_funcs.get_run_funcs()
+
+    def __init__(self, param_json, sim_type, data_gen_type):
+        self._params = json.loads(param_json)
+        self._run(sim_type, data_gen_type)
+
+    def _run(self, sim_type, data_gen_type):
+        sim_run_func, sim_prereq_params = Runner.run_func_ltable[sim_type]
+        gen_run_func, gen_prereq_params = Runner.data_gen_func_ltable[data_gen_type]
+
+        if not self._sim_has_required_params_for_given_run_func(
+            sim_prereq_params, sim_type
+        ):
+            return
+
+        if not self._sim_has_required_params_for_given_run_func(
+            gen_prereq_params, data_gen_type
+        ):
+            return
+
+        print('Generating "{}" data...'.format(data_gen_type))
```

Comment:
The `_run` method seems to have two concerns.  The parts above are initialization related, and the code below is calling `gen_run_func` and `sim_run_func` are actually 'running' the simulation.  Let's split this up so that we have everything initialized at Runner construction time and then executed when run() is invoked.